pulser-core/pulser/sampler/samples.py

```diff
@@ -292,6 +292,10 @@ class SequenceSamples:
     samples_list: list[ChannelSamples]
     _ch_objs: dict[str, Channel]
     _slm_mask: _SlmMask = field(default_factory=_SlmMask)
+    _interaction_coeff_xy: float | None = None
+    _interaction_coeff: float | None = None
```

Comment:
2 other approaches here:
- store the device of the `Sequence`
- provide the device in the init of the Simulation (I favored keeping the signature as simple as possible)